flag_map_impl.go

```diff
@@ -10,9 +10,12 @@ import (
 
 // MapBase wraps map[string]T to satisfy flag.Value
 type MapBase[T any, C any, VC ValueCreator[T, C]] struct {
-	dict       *map[string]T
-	hasBeenSet bool
-	value      Value
+	dict                  *map[string]T
+	hasBeenSet            bool
+	value                 Value
+	sliceSeparator        string
```

Comment:
```suggestion
	multiValueConfig multiValueParsingConfig
```

That way we dont need to update individual structs if add more fields for this